xpti/include/xpti/xpti_trace_framework.hpp

```diff
@@ -479,84 +559,930 @@ inline std::string readMetadata(const metadata_t::value_type &MD) {
          std::to_string(RawData.size);
 }
 
+/// @brief Creates a unique 128-bit identifier (UID) for tracking entities.
+///
+/// This function combines file and function identifiers with line and column
+/// information to generate a unique identifier. The UID is composed of two
+/// 64-bit parts: the first part (p1) combines the file and function IDs, and
+/// the second part (p2) combines the column and line numbers. An initial
+/// instance count of 1 is set, indicating the creation of a new UID.
+///
+/// @param FileID The unique identifier for the file.
+/// @param FuncID The unique identifier for the function.
+/// @param Line The line number where the entity is located.
+/// @param Col The column number where the entity is located.
+/// @return A `xpti::uid128_t` structure representing the unique identifier.
+///
+inline xpti::uid128_t make_uid128(uint64_t FileID, uint64_t FuncID, int Line,
+                                  int Col) {
+  xpti::uid128_t UID;
+  UID.p1 = (FileID << 32) | FuncID;
+  UID.p2 = ((uint64_t)Col << 32) | Line;
+  UID.instance = 0;
+  return UID;
+}
+
+/// @brief Checks if a given 128-bit UID is valid.
+///
+/// A 128-bit UID is considered valid if neither of its parts (p1, p2) are zero
+/// and its instance number is greater than 0. This function evaluates these
+/// conditions and returns true if all are met, indicating the UID is valid.
+///
+/// @param UID The 128-bit UID to be checked.
+/// @return True if the UID is valid, false otherwise.
+///
+inline bool is_valid_uid(const xpti::uid128_t &UID) {
```

Comment:
I see this check is included in tracepoint isValid checks. p[robably would be better to reuse it there

Reply:
Fixed!